### DISCUSSION TITLE
Can I support jest.fn() etc. by adding `define: { jest: 'vi' }` in vite.config.ts ?

### DISCUSSION THREAD

**Discussion by guoyunhe** [vitest-dev/vitest#5243] [Q&A]
I have some old Jest projects that I want to migrate to Vitest. These projects contain a lot of `jest.fn()`, `jest.useFakeTimers()`, etc. I see that vitest provides similar functions `vi.fn()`, `vi.useFakeTimers()`. So I think if I can simply alias `jest` to `vi`:

```js
defineConfig({
  define: {
    jest: 'vi',
  },
});
```

Is it a reliable replacement?

**Comment by hi-ogawa** [comment]
`define` based replacement might actually work (assuming you have `globals` enabled https://vitest.dev/config/#globals), but I think what I would try first is to do something like this in `setupFiles` https://vitest.dev/config/#setupfiles and see which test fails:

```ts
import { vi } from "vitest";

Object.assign(globalThis, { jest: vi })
```

I remember I did something similar when migrating from Mocha since mocha uses `before/after` but Jest/Vitest uses `beforeAll/afterAll`.
